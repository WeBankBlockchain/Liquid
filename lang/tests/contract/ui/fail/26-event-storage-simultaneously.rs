use liquid_lang as liquid;

#[liquid::contract]
mod noop {
    #[liquid(event)]
    #[liquid(storage)]
    struct Noop {}

    #[liquid(methods)]
    impl Noop {
        pub fn new(&mut self) {}

        pub fn noop(&self) {}
    }
}

fn main() {}
