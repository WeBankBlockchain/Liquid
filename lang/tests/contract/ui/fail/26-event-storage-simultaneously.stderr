error: a struct can be marked by only one of the followings: `liquid(storage)`, `liquid(event)` or `liquid(asset)` at the same time
 --> $DIR/26-event-storage-simultaneously.rs:5:5
  |
5 | /     #[liquid(event)]
6 | |     #[liquid(storage)]
7 | |     struct Noop {}
  | |__________________^
